archeryutils/handicaps/handicap_scheme.py

```diff
@@ -34,14 +34,37 @@
 import itertools as itr
 import warnings
 from abc import ABC, abstractmethod
-from typing import TypeVar, cast, overload
+from typing import cast
 
 import numpy as np
 import numpy.typing as npt
 
 from archeryutils import rounds, targets
 
-FloatArray = TypeVar("FloatArray", float, npt.NDArray[np.float64])
+
+def _cast_float_array(var_in: npt.ArrayLike) -> npt.NDArray[np.float64]:
+    """Check we can cast to np.float64 array and do so.
+
+    Parameters
+    ----------
+    var_in : ArrayLike
+        input that is an ArrayLik
+
+    Returns
+    -------
+    NDArray[np.float64]
+        var_in cast to NDArray[np.float64]
+
+    Raises
+    ------
+    TypeError
+        If an ArrayLike that cannot be cast is passed in.
+
+    """
+    if np.can_cast(np.asarray(var_in), np.float64):
+        return np.asarray(var_in).astype(np.float64, casting="safe")
+    err_msg = f"Inappropriate input for handicaps code. Must be numeric value."
+    raise TypeError(err_msg)
```

Comment:
Rather than explicitly checking `can_cast`, I'd normally just use `np.asarray(..., dtype=float)` and do it all in one step. I think that could replace the use of the `_cast_float_array` function altogether, though it might be worth keeping around if you specifically need to raise a `TypeError` on failure (weirdly, `np.asarray(...)` raises a `ValueError` if the types can't be converted)

Reply:
Simplified in https://github.com/jatkinson1000/archeryutils/pull/133/commits/954a89bbba53c483db18b5cabd1eb91fac2ad010 but decided to keep the function to get the friendlier type error given it triggers quite a way down a call stack.